bindings/java/src/main/native/jucx_common_def.cc

```diff
@@ -173,10 +176,15 @@ static inline void set_jucx_request_completed(JNIEnv *env, jobject jucx_request,
                                               struct jucx_context *ctx)
 {
     env->SetObjectField(jucx_request, native_id_field, NULL);
-    if ((ctx != NULL) && (ctx->length > 0)) {
-        env->SetLongField(jucx_request, recv_size_field, ctx->length);
-    } else if ((ctx != NULL) && (ctx->iovec != NULL)) {
-        ucs_free(ctx->iovec);
+    if (ctx != NULL) {
+        env->SetLongField(jucx_request, sender_tag_field, ctx->sender_tag);
+        if (ctx->length > 0) {
+            env->SetLongField(jucx_request, recv_size_field, ctx->length);
+        }
```

Comment:
why only if length>0?

Reply:
length is initialized to 0. This method used to complete not only recv requests (send/one-sided, etc.).

Reply:
what if receive completes with length 0?

Reply:
In java long initializes as 0. So by default `ucpRequest.recvSize == 0` and `ucpRequest.senderTag == 0`

Reply:
what is different between senderTag and recvSize, that ctx->sender_tag is not checked but ctx->length is ?